nvflare/app_common/pt/log_receiver.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+from logging import LogRecord
+from typing import List, Optional
+
+from nvflare.apis.analytix import AnalyticsData, AnalyticsDataType
+from nvflare.apis.dxo import DXO, from_shareable
+from nvflare.apis.fl_constant import LogMessageTag
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.shareable import Shareable
+from nvflare.app_common.widgets.streaming import AnalyticsReceiver
+
+
+def _get_sender_log_info(dxo: DXO, record_origin: str) -> str:
+    record: LogRecord = dxo.data.get(LogMessageTag.LOG_RECORD)
+    data = {
+        "from-client": record_origin,
+        "name": record.name,
+        "log_level": record.levelname,
+        "pathname": record.pathname,
+        "lineno": record.lineno,
+        "func": record.funcName,
+        "msg": record.msg,
+    }
+    return json.dumps(data)
+
+
+class LogAnalyticsReceiver(AnalyticsReceiver):
```

Comment:
The logic looks good to me.

But this LogAnalyticsReceiver is general purpose and it has no dependencies of PyTorch.
So it should be put in a different place.

Suggest just put in "nvflare/app_common/widgets/" folder.

Reply:
good point. Moved to "nvflare/app_common/widgets/".